From e8682101f1f73222cf53df77177358b2030cde61 Mon Sep 17 00:00:00 2001
From: Lily Zhang <r58066@freescale.com>
Date: Mon, 8 Aug 2011 17:37:47 +0800
Subject: [PATCH] ENGR00155282 mc34708: support to build mc34708 driver only

1. If change defconfig and only build in mc34708 dirver,
the following error is reported:
undefined reference to `mc13892_alloc_data'
undefined reference to `mc13892_init_registers'
undefined reference to `mc13892_get_revision'

This patch is to fix above compiling error.

2. Add mc34708 dependency in Kconfig

Signed-off-by: Lily Zhang <r58066@freescale.com>
---
 drivers/mxc/pmic/Kconfig              |    4 ++--
 drivers/mxc/pmic/core/pmic_core_i2c.c |    8 ++++++++
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/drivers/mxc/pmic/Kconfig b/drivers/mxc/pmic/Kconfig
index 6f69ab0..2058b6b 100644
--- a/drivers/mxc/pmic/Kconfig
+++ b/drivers/mxc/pmic/Kconfig
@@ -33,11 +33,11 @@ config MXC_PMIC_MC34708
 
 config MXC_PMIC_I2C
 	bool "Support PMIC I2C Interface"
-	depends on MXC_PMIC_MC13892 && I2C
+	depends on (MXC_PMIC_MC13892 || MXC_PMIC_MC34708) && I2C
 
 config MXC_PMIC_SPI
 	bool "Support PMIC SPI Interface"
-	depends on (MXC_PMIC_MC13892 || MXC_PMIC_MC13783) && SPI
+	depends on (MXC_PMIC_MC13892 || MXC_PMIC_MC13783 || MXC_PMIC_MC34708) && SPI
 
 config MXC_PMIC_MC34704
 	tristate "MC34704 PMIC"
diff --git a/drivers/mxc/pmic/core/pmic_core_i2c.c b/drivers/mxc/pmic/core/pmic_core_i2c.c
index 77530c4..fae41bb 100644
--- a/drivers/mxc/pmic/core/pmic_core_i2c.c
+++ b/drivers/mxc/pmic/core/pmic_core_i2c.c
@@ -104,14 +104,22 @@ static struct platform_device pwm2_ldm = {
 };
 
 enum pmic_id {
+#if defined(CONFIG_MXC_PMIC_MC13892) || defined(CONFIG_MXC_PMIC_MC13892_MODULE)
 	PMIC_ID_MC13892,
+#endif
+#if defined(CONFIG_MXC_PMIC_MC34708) || defined(CONFIG_MXC_PMIC_MC34708_MODULE)
 	PMIC_ID_MC34708,
+#endif
 	PMIC_ID_INVALID,
 };
 
 static struct pmic_internal pmic_internal[] = {
+#if defined(CONFIG_MXC_PMIC_MC13892) || defined(CONFIG_MXC_PMIC_MC13892_MODULE)
 	[PMIC_ID_MC13892] = _PMIC_INTERNAL_INITIALIZER(mc13892),
+#endif
+#if defined(CONFIG_MXC_PMIC_MC34708) || defined(CONFIG_MXC_PMIC_MC34708_MODULE)
 	[PMIC_ID_MC34708] = _PMIC_INTERNAL_INITIALIZER(mc34708),
+#endif
 };
 
 static int get_index_pmic_internal(const char *name)
-- 
1.5.4.4

